Model/DeletedProductData.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Metrilo\Analytics\Model;
+
+class DeletedProductData
+{
+    public function __construct(
+        \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderCollection
+    ) {
+        $this->orderCollection = $orderCollection;
+    }
+    
+    public function getDeletedProducts($storeId) {
+        $query = "SELECT sales_order_item.item_id, sales_order_item.parent_item_id, sales_order_item.product_id,
+                 sales_order_item.store_id, sales_order_item.product_type, sales_order_item.name,
+                 sales_order_item.sku, sales_order_item.price, sales_order_item.order_id
+          FROM `sales_order_item`
+          LEFT OUTER JOIN `catalog_product_entity`
+          ON sales_order_item.product_id = catalog_product_entity.entity_id
+          WHERE catalog_product_entity.entity_id IS NULL
+          AND sales_order_item.store_id = '$storeId'";
```

Comment:
I'd really think twice before concatenating variables into MySQL query without any escaping. You're possibly creating an injection vulnerability.

As I see, `$storeId` comes from a request parameter:
```php
$storeId           = (int)$this->request->getParam('storeId');
```
[Controller/Adminhtml/Import/Ajax.php:55](https://github.com/Metrilo/metrilo-magento2-integration/blob/26104e328fdcfd0c757debbb4ae1020b8fbc17d6/Controller/Adminhtml/Import/Ajax.php#L55)

Reply:
Excellent point.

Reply:
Will address that issue by reworking the raw sql query to use the m2 ORM, this will prevent any sql injections accessing the database.